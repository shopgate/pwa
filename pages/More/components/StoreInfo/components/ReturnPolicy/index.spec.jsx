import React from 'react';
import { shallow } from 'enzyme';

/**
 * Creates a component for testing.
 * @param {boolean} mockedShow Is the return policy supposed to be shown.
 * @returns {Object}
 */
const createComponent = (mockedShow) => {
  jest.resetModules();
  jest.mock('@shopgate/pwa-common-commerce/market/helpers/showReturnPolicy', () => mockedShow);
  const ReturnPolicy = require.requireActual('./index').default;

  return shallow(<ReturnPolicy />);
};

describe('<ReturnPolicy />', () => {
  it('should render as expected when the return policy is supposed to be shown', () => {
    const wrapper = createComponent(true);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MoreMenuItem').exists()).toBe(true);
    expect(wrapper.find('Portal')).toHaveLength(3);
  });

  it('should render as expected when the return policy is not supposed to be shown', () => {
    const wrapper = createComponent(false);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MoreMenuItem').exists()).toBe(false);
    expect(wrapper.find('Portal')).toHaveLength(2);
  });
});
