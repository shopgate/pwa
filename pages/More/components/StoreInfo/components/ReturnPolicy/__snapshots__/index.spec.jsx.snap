// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReturnPolicy /> should render as expected when the return policy is not supposed to be shown 1`] = `
<Fragment>
  <Portal
    name="nav-menu.return-policy.before"
    props={
      Object {
        "Headline": [Function],
        "Item": [Function],
        "Section": [Function],
      }
    }
  />
  <Portal
    name="nav-menu.return-policy.after"
    props={
      Object {
        "Headline": [Function],
        "Item": [Function],
        "Section": [Function],
      }
    }
  />
</Fragment>
`;

exports[`<ReturnPolicy /> should render as expected when the return policy is supposed to be shown 1`] = `
<Fragment>
  <Portal
    name="nav-menu.return-policy.before"
    props={
      Object {
        "Headline": [Function],
        "Item": [Function],
        "Section": [Function],
      }
    }
  />
  <Portal
    name="nav-menu.return-policy"
    props={
      Object {
        "Headline": [Function],
        "Item": [Function],
        "Section": [Function],
      }
    }
  >
    <MoreMenuItem
      className={null}
      href="/page/return_policy"
      label="navigation.return_policy"
      onClick={null}
      testId={null}
    />
  </Portal>
  <Portal
    name="nav-menu.return-policy.after"
    props={
      Object {
        "Headline": [Function],
        "Item": [Function],
        "Section": [Function],
      }
    }
  />
</Fragment>
`;
